src/vs/workbench/contrib/positronNotebook/browser/PositronNotebookInstance.ts

```diff
@@ -1064,6 +1098,38 @@ export class PositronNotebookInstance extends Disposable implements IPositronNot
 	// =============================================================================================
 	// #region Private Methods
 
+	/**
+	 * Computes the minimal splice operations needed to transform oldCells into newCells.
+	 * Returns an array of NotebookViewCellsSplice tuples [start, deleteCount, insertedCells].
+	 * @param oldCells The previous cell array
+	 * @param newCells The new cell array
+	 * @returns Array of splice operations
+	 */
+	private _computeCellSplices(oldCells: IPositronNotebookCell[], newCells: IPositronNotebookCell[]): ContextKeysNotebookViewCellsSplice[] {
+		// Find first difference
+		let start = 0;
+		while (start < oldCells.length && start < newCells.length && oldCells[start] === newCells[start]) {
+			start++;
+		}
+
+		// Find last difference
+		let oldEnd = oldCells.length;
+		let newEnd = newCells.length;
+		while (oldEnd > start && newEnd > start && oldCells[oldEnd - 1] === newCells[newEnd - 1]) {
+			oldEnd--;
+			newEnd--;
+		}
+
+		const deleteCount = oldEnd - start;
+		const inserted = newCells.slice(start, newEnd);
+
+		if (deleteCount === 0 && inserted.length === 0) {
+			return [];
+		}
+
+		return [[start, deleteCount, inserted]];
+	}
```

Comment:
Can you think of a time where we may want to convert this do find multiple splices? Or are the type of cell operations always going to be simple enough that it's never worth it to get more complicated?

e.g. 

```suggestion
	 * Returns an array of NotebookViewCellsSplice tuples [start, deleteCount, insertedCells].
	 * @param oldCells The previous cell array
	 * @param newCells The new cell array
	 * @returns Array of splice operations
	 */
	private _computeCellSplices(oldCells: IPositronNotebookCell[], newCells: IPositronNotebookCell[]): ContextKeysNotebookViewCellsSplice[] {
		// Create a Set for quick contains checking
		const oldCellsSet = new Set(oldCells);

		// Use the diff algorithm to compute multiple splices for non-contiguous changes
		// import {..., diff } from '../../notebook/common/notebookCommon.js'
		const splices = diff(
			oldCells,
			newCells,
			(cell) => oldCellsSet.has(cell), 
			(a, b) => a === b 
		);

		return splices.map(splice => [splice.start, splice.deleteCount, [...splice.toInsert]]);
	}

```